types/amplitude-js/index.d.ts

```diff
@@ -1,61 +1,146 @@
-// Type definitions for Amplitude SDK 2.12.1
+// Type definitions for Amplitude SDK 4.4.0
 // Project: https://github.com/amplitude/Amplitude-Javascript
 // Definitions by: Arvydas Sidorenko <https://github.com/Asido>
-// Definitions: https://github.com/Asido/DefinitelyTyped
+//                 Dan Manastireanu <https://github.com/danmana>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 declare module amplitude {
+
+    type Callback = (responseCode: number, responseBody: string, details?: { reason: string; }) => void;
```

Comment:
This doesn't match https://amplitude.zendesk.com/hc/en-us/articles/115002889587#type-definitions is the third parameter intentional?

Reply:
Indeed it doesn't match the public documentation, but the third parameter is passed in code (in most cases contains some detailed description of an error):

In the following examples, the details.reason is the only way to know which is the reason for the error.

If you think this exposes too much of the internal implementation I can remove it.

https://github.com/amplitude/Amplitude-JavaScript/blob/master/src/amplitude-client.js#L826
https://github.com/amplitude/Amplitude-JavaScript/blob/master/src/amplitude-client.js#L844
https://github.com/amplitude/Amplitude-JavaScript/blob/master/src/amplitude-client.js#L850